pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -522,6 +544,46 @@ func (h *NetworkUtilsHandler) DisableTXOffloadChecksum(ifaceName string) error {
 	return nil
 }
 
+func (h *NetworkUtilsHandler) CreateRADaemon(socketPath string, advertisementIfaceName string, ipv6CIDR string, currentRetry int) error {
```

Comment:
I would move most of the code in this method to pkg/virt-launcher/virtwrap/network/ndp/connection.go. And invoke from this method only-
1. ndp.Import (containing all the code till the go routine).
2. ndp.RunRAServer ()

And also would change it's name to `CreateRaServer`.